packages/node/src/transport/WsInboundTransport.ts

```diff
@@ -83,18 +82,26 @@ export class WebSocketTransportSession implements TransportSession {
   public id: string
   public readonly type = 'WebSocket'
   public socket: WebSocket
+  private logger!: Logger
 
-  public constructor(id: string, socket: WebSocket) {
+  public constructor(id: string, socket: WebSocket, logger: Logger) {
     this.id = id
     this.socket = socket
+    this.logger = logger
   }
 
   public async send(agentContext: AgentContext, encryptedMessage: EncryptedMessage): Promise<void> {
     if (this.socket.readyState !== WebSocket.OPEN) {
       throw new AriesFrameworkError(`${this.type} transport session has been closed.`)
     }
-
-    this.socket.send(JSON.stringify(encryptedMessage))
+    this.socket.send(JSON.stringify(encryptedMessage), (error?) => {
+      if (error != undefined) {
+        this.logger.error('Error sending message: ' + error)
+        throw new AriesFrameworkError(`${this.type} send message failed.`)
+      } else {
+        this.logger.debug(`${this.type} sent message successfully.`)
+      }
```

Comment:
Wouldn't be better to just throw an error here and evaluate it where we call `await session.send(agentContext, encryptedMessage)`?

Reply:
Not sure what 'better' means, personally I prefer having debug logs as close as possible to where things happen, and from the existing logs it was impossible to decide whether a message had been sent successfully or not...

Reply:
Have in mind that this is a debug log, typically activated only when troubleshooting, so you'd want to know where exactly things go wrong

Reply:
Slightly changed to bubble up the error

Reply:
Do you know if `socket.send()` blocks until the callback is actually called? 

I think it will be nice to return from `session.send()` only if it could actually write successfully to the socket. Not sure how it works, but probably we'll need to wait until the callback is called and check the error code before returning.

Reply:
I'm sorry, I should've clarified it in my first comment. The idea is that the `WebSocketSession` is more an entity or value object than a service object. Therefore I would rather avoid passing the logger there. In the same way, we don't pass (hopefully) the logger into records.

Reply:
Oh my bad, I should have asked before merging. But don't worry, there will be more changes in these classes shortly, so we can take the opportunity to analyze the flows where session.send is called and call the logger from an upper layer if needed.